tests/unit/model/test_software_system.py

```diff
@@ -18,10 +18,32 @@
 
 import pytest
 
-from structurizr.model.model import Model
+from structurizr.model.container import Container
 from structurizr.model.software_system import SoftwareSystem
 
 
+class MockModel:
+    """Implement a mock model for testing."""
+
+    def add_container(self, container):
+        """Simulate the model assigning IDs to new elements."""
+        if not container.id:
+            container.id = "id"
+        container.set_model(self)
+        pass
+
+
+_model = MockModel()
```

Comment:
I didn't fully understand your earlier comment about this being garbage collected. The way I'd do it is:

```py
@pytest.fixture(scope="function")
def model():
   return MockModel()

@pytest.fixture(scope="function")
def model_with_system(model):
    software_system = SoftwareSystem(name="Sys")
    software_system.set_model(model)
    return model


def test_software_system_add_constructed_container(model_with_system):
    system = model.get_system...
    ...
```

Plus the docstrings :wink:

And possibly a separate fixture like you currently have but without a model set on it for some of the tests.

Reply:
What I found is that unit tests failed unexpectedly because the Element only holds a weak ref to the model so by the time we needed it then the model had been collected.  I think your suggestion would have the same problem.

Reply:
Here I am suggesting to pass the model down to the test function and the model holds the system. So it should not be garbage collected.

Reply:
Ah, sorry - I'd not spotted that the test methods were now taking the model rather than the system.  I'll have a look at that.  Thanks.

Reply:
OK, I've given that a go.  See what you reckon.